core/services/blockhashstore/coordinators.go

```diff
@@ -76,53 +89,89 @@ func (v *V1Coordinator) Requests(
 	fromBlock uint64,
 	toBlock uint64,
 ) ([]Event, error) {
-	iter, err := v.c.FilterRandomnessRequest(&bind.FilterOpts{
-		Start:   fromBlock,
-		End:     &toBlock,
-		Context: ctx,
-	}, nil)
+	logs, err := v.lp.LogsWithSigs(
+		int64(fromBlock),
+		int64(toBlock),
+		[]common.Hash{
+			v1.VRFCoordinatorRandomnessRequest{}.Topic(),
+		},
+		v.c.Address(),
+		pg.WithParentCtx(ctx))
 	if err != nil {
-		return nil, errors.Wrap(err, "filter v1 requests")
+		return nil, errors.Wrap(err, "fetching VRF requests")
 	}
-	defer iter.Close()
+
 	var reqs []Event
-	for iter.Next() {
-		reqs = append(reqs, Event{
-			ID:    hex.EncodeToString(iter.Event.RequestID[:]),
-			Block: iter.Event.Raw.BlockNumber,
-		})
+	for _, l := range logs {
+		requestLog, err := v.c.ParseLog(l.ToGethLog())
+		if err != nil {
+			continue // malformed log should not break flow
+		}
+		request, ok := requestLog.(*v1.VRFCoordinatorRandomnessRequest)
+		if !ok {
+			continue // malformed log should not break flow
+		}
+		reqs = append(reqs, Event{ID: hex.EncodeToString(request.RequestID[:]), Block: request.Raw.BlockNumber})
 	}
+
 	return reqs, nil
 }
 
 // Fulfillments satisfies the Coordinator interface.
 func (v *V1Coordinator) Fulfillments(ctx context.Context, fromBlock uint64) ([]Event, error) {
-	iter, err := v.c.FilterRandomnessRequestFulfilled(&bind.FilterOpts{
-		Start:   fromBlock,
-		Context: ctx,
-	})
+	toBlock, err := v.lp.LatestBlock()
 	if err != nil {
-		return nil, errors.Wrap(err, "filter v1 fulfillments")
+		return nil, errors.Wrap(err, "fetching latest block")
 	}
-	defer iter.Close()
+
+	logs, err := v.lp.LogsWithSigs(
+		int64(fromBlock),
+		int64(toBlock),
```

Comment:
I just realized that we don't have an LP query which is equivalent to the above `FilterRandomnessRequestFulfilled` query (i.e `toBlock` unspecified). Maybe this is okay though.